tests/test_models.py

```diff
@@ -432,6 +432,18 @@ def test_score_t_with_score_hrt_equality(self) -> None:
 
         assert torch.allclose(scores_t, scores_hrt, atol=1e-06)
 
+    def test_reset_parameters_constructor_call(self):
+        """Tests whether reset_parameters is called in the constructor."""
+        self.model.reset_parameters_ = None
+        try:
+            self.model.__init__(
+                self.factory,
+                embedding_dim=self.embedding_dim,
+                **(self.model_kwargs or {})
+            )
+        except TypeError as error:
```

Comment:
Alternative, using unittests:

```python
        with self.assertRaises(TypeError) as cm:
            self.model.__init__(
                self.factory,
                embedding_dim=self.embedding_dim,
                **(self.model_kwargs or {})
            )
            self.assertEqual(cm.exception.args[0], "'NoneType' object is not callable")
```

Reply:
Any preferences?

Reply:
i like using unittests since the code gives additional information about why you're doing this

Reply:
I just checked the alternative: The assertEqual is not evaluated since the exception occurs before. So I just leave it as it is.

Reply:
I must have done it wrong, I think it should have come after the with statement (which manages to catch the exception quite opaquely, using magic)